### PR TITLE
project, service: move version bundles content to pkg/project

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/7722

This is kind of weird because of multiple providers. But this is the problem of
this particular branch with the design we already abandoned so I don't want to
overthink that.

I don't want to keep version_bundle.go files in the `controller` package 
because they are not used there.

### CODE REVIEW

File: pkg/project/version_bundle_aws.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package project
+
+import (
+	"github.com/giantswarm/versionbundle"
+)
+
+var versionBundleAWS = versionbundle.Bundle{
```
Comment (kopiczko) [inline pkg/project/version_bundle_aws.go]: This is only extracted from service/controller/aws/version_bundle.go.

File: pkg/project/version_bundle_azure.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package project
+
+import (
+	"github.com/giantswarm/versionbundle"
+)
+
+var versionBundleAzure = versionbundle.Bundle{
```
Comment (kopiczko) [inline pkg/project/version_bundle_azure.go]: This is only extracted from service/controller/azure/version_bundle.go.

File: pkg/project/version_bundle_kvm.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package project
+
+import (
+	"github.com/giantswarm/versionbundle"
+)
+
+var versionBundleKVM = versionbundle.Bundle{
```
Comment (kopiczko) [inline pkg/project/version_bundle_kvm.go]: This is only extracted from service/controller/kvm/version_bundle.go.

File: service/service.go [hunk 4]
```diff
@@ -282,7 +283,7 @@ func New(config Config) (*Service, error) {
 			Name:           config.ProjectName,
 			Source:         config.Source,
 			Version:        config.Version,
-			VersionBundles: NewVersionBundles(provider),
+			VersionBundles: project.NewVersionBundles(),
```
Comment (kopiczko) [inline service/service.go]: `provider` was not used so I dropped it from the signature.

File: service/controller/aws/resource_set.go [hunk 5]
```diff
@@ -419,11 +420,11 @@ func newResourceSet(config resourceSetConfig) (*controller.ResourceSet, error) {
 			return false
 		}
 
-		if key.OperatorVersion(&cr) == VersionBundle().Version {
+		if key.OperatorVersion(&cr) == project.BundleVersion() {
```
Comment (xh3b4sd) [inline service/controller/aws/resource_set.go]: When this is `legacy` it should not be concerned with the version label I think. We only introduced this for CAPI.

Comment (kopiczko) [reply service/controller/aws/resource_set.go]: We should have it for legacy objects as well. We'll need that for label selector. Whatever is flat must  use it eventually.

Comment (xh3b4sd) [reply service/controller/aws/resource_set.go]: Why is that then only in the AWS resource set? Then I would expect this to also be true for Azure and KVM.

Comment (kopiczko) [reply service/controller/aws/resource_set.go]: You're right here.

Comment (kopiczko) [reply service/controller/aws/resource_set.go]: I made proper notes. We'll sort that later.

**Pull request merged**